### ISSUE TITLE
Support for unchecked exceptions

### ISSUE THREAD

**Issue by breucode** [TNG/ArchUnit#886]
I would like to check, if a method throws an unchecked exception.

Currently, it seems, we can only check, if such Exceptions are instantiated, but there is no way, to look for a "throw" invocation.

**Comment by mslowiak** [comment]
Hi there,
I would like to dig into that issue, but I need some guidance :) 
Where I can find a code connected to `throws statement is not supported yet, **only throws declarations and catch clauses**`?
Should the new logic resist in the same place?

**Comment by codecholeric** [comment]
Hi @mslowiak, that's great! But just a warning, this might be a challenging issue :wink: (in other words, there are probably lower hanging fruits within the other open issues)

But if you're up for a challenge, the point to look into would be `com.tngtech.archunit.core.importer.JavaClassProcessor`. That is the place that hooks into the ASM API to parse the bytecode. The code for throws declarations is e.g. in `com.tngtech.archunit.core.importer.JavaClassProcessor#visitMethod` and the code for try-catch-blocks in `com.tngtech.archunit.core.importer.JavaClassProcessor.MethodProcessor#visitTryCatchBlock`.
Let me know if you have any further questions!

**Comment by mslowiak** [comment]
I have one more question regarding _But the tricky part would likely be to find out which type of exception will actually be thrown, because that depends on the current stack_.

Did you meant that if exception is thrown but then wrapped into some try catch then it shouldn't be counted as a violation of this rule? Or you had something different in mind?

**Comment by codecholeric** [comment]
No, what I meant is that in the place where you see that an exception is thrown (`MethodProcessor.visitInsn(int opCode)`) you don't see which exception is actually thrown, but only that an `ATHROW` instruction is in the bytecode. So you need to figure out how to get the actual exception once you encounter this opcode.
